src/GraphQL.Conventions/Adapters/Engine/GraphQLEngine.cs

```diff
@@ -308,14 +316,15 @@ public GraphQLExecutor NewExecutor(IRequestDeserializer requestDeserializer = nu
                 configuration.Listeners.Add(new DataLoaderListener());
             }
 
+            //TODO : refactor todo in schemabuilder
```

Comment:
What's remaining here?

Reply:
Now the middlewares are setup on the schema itself, so it's probably not the best place to register them in the execute method. I understood from the GraphQL migration guide that once a request is executed on the schema it's not possible to add/remove middlewares, so if you want to reuse the same GraphQLEngine it need to be more clear that the middlewares need to be set before the first Execute call. Need to have a deeper look at the GraphQL implementation.

Reply:
Ah, yes. Gotcha.

Reply:
My 50 cents - you _can_ change schema after its initialization (resolvers, descriptions, just anything) but must synchronize changes yourself. Also `SchemaVisitor` can work like field middleware, doing event more things, not only replacing/wrapping field resolver.

Reply:
Fair. Do you think it's worth adding complexity for it, though? Any obvious use cases spring to mind? :)

Reply:
Schema visitors work well with applied directives.